third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1074,9 +1074,52 @@ struct FDivOpConversion
                                    ConversionPatternRewriter &rewriter,
                                    Type elemTy, MultipleOperandsRange operands,
                                    Location loc) const {
+    // For non-F32 input, it's lowered to LLVM::FDivOp, which is a
+    // IEEE-compliant DIV operation.
+    if (elemTy.getIntOrFloatBitWidth() != 32)
+      return {rewriter.create<LLVM::FDivOp>(loc, elemTy, operands[0][0],
+                                            operands[0][1])};
 
-    return {rewriter.create<LLVM::FDivOp>(loc, elemTy, operands[0][0],
-                                          operands[0][1])};
+    // The algorithm comes from
+    // https://github.com/llvm/llvm-project/blob/bda7aadfcd0596675c8b4b381f178f3d8f854f20/llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp#L4980-L5065
+    // with the Newton-Raphson refinement removed, to perform a faster,
+    // approximated DIV operation, aligning with the `div.full.f32` instruction
+    // on the NV backend.
+    Value &lhs = operands[0][0];
+    Value &rhs = operands[0][1];
+    MLIRContext *ctx = rewriter.getContext();
+    Type divScaleResType = struct_ty({elemTy, i1_ty});
+
+    // The `llvm.amdgcn.div.scale.f32` instruction has two outputs: the scaled
+    // variable and the VCC register.
+    LLVM::CallIntrinsicOp denominatorScaleOp = LLVM::createLLVMIntrinsicCallOp(
```

Comment:
Style nit: when it's very clear what the result type would be (like here from the `LLVM::createLLVMIntrinsicCallOp` func name), you can use `auto` as the result type.